lib/report/k8s.go

```diff
@@ -78,7 +80,7 @@ func capturePreviousContainerLogs(ctx context.Context, namespaces []string, runn
 				// Collect logs for the previous instance of the container if there's any.
 				name := fmt.Sprintf("k8s-logs-%v-%v-%v-prev", namespace, pod, container)
 				collectors = append(collectors, Cmd(name, kubectl.Command("logs", pod,
-					"--namespace", namespace, "-p",
+					"--namespace", namespace, "-p", "--since", since.String(),
```

Comment:
This only filters the k8s logs for _previous_ instances of containers (`-p` flag). There's a cluster-info call above that still collects all logs for all currently running containers. To filter those too, maybe we can skip collecting "cluster-info" if time filter is provided, and instead use the same "kubectl logs" command as here but without "-p" to collect logs of current containers with proper time filter applied?

Reply:
Kubernetes report structure. `kubectl cluster-info dump` collects a bit more than just the logs of each pod. Do we want to manually collect this information as well when the `--since` flag is provided?

* `--since` not provided
```
├── default
│   ├── daemonsets.json
│   ├── deployments.json
│   ├── events.json
│   ├── pods.json
│   ├── replicasets.json
│   ├── replication-controllers.json
│   └── services.json
├── k8s-logs-kube-system-coredns-9rr89-coredns-prev
├── k8s-logs-kube-system-coredns-g9xv5-coredns-prev
├── k8s-logs-kube-system-coredns-sthwk-coredns-prev
├── k8s-logs-kube-system-site-app-post-install-2bcd3f-79kft-post-install-hook-prev
├── k8s-logs-monitoring-influxdb-78f4754b8f-zql59-watcher-prev
├── kube-node-lease
│   ├── daemonsets.json
│   ├── deployments.json
│   ├── events.json
│   ├── pods.json
│   ├── replicasets.json
│   ├── replication-controllers.json
│   └── services.json
├── kube-public
│   ├── daemonsets.json
│   ├── deployments.json
│   ├── events.json
│   ├── pods.json
│   ├── replicasets.json
│   ├── replication-controllers.json
│   └── services.json
├── kube-system
│   ├── bandwagon-6c4b5b5c76-tf9kc
│   │   └── logs.txt
│   ├── bandwagon-install-ab9d29-779nz
│   │   └── logs.txt
│   ├── coredns-9rr89
│   │   └── logs.txt
│   ├── coredns-g9xv5
│   │   └── logs.txt
│   ├── coredns-sthwk
│   │   └── logs.txt
│   ├── daemonsets.json
│   ├── deployments.json
│   ├── dns-app-install-c4f31f-j2cp5
│   │   └── logs.txt
│   ├── events.json
│   ├── gravity-install-fa4102-hkpvv
│   │   └── logs.txt
│   ├── gravity-site-c7jxz
│   │   └── logs.txt
│   ├── gravity-site-xlrwl
│   │   └── logs.txt
│   ├── gravity-site-zg7j5
│   │   └── logs.txt
│   ├── log-collector-697d94486-vftvx
│   │   └── logs.txt
│   ├── log-forwarder-5rqnm
│   │   └── logs.txt
│   ├── log-forwarder-g7pk2
│   │   └── logs.txt
│   ├── log-forwarder-vbt9r
│   │   └── logs.txt
│   ├── logging-app-bootstrap-fec658-qr7q4
│   │   └── logs.txt
│   ├── monitoring-app-install-f4a4b8-mckh4
│   │   └── logs.txt
│   ├── pods.json
│   ├── replicasets.json
│   ├── replication-controllers.json
│   ├── services.json
│   ├── site-app-post-install-2bcd3f-79kft
│   │   └── logs.txt
│   ├── tiller-app-bootstrap-cb1643-7bdwt
│   │   └── logs.txt
│   └── tiller-deploy-7997c85989-9f856
│       └── logs.txt
├── monitoring
│   ├── daemonsets.json
│   ├── deployments.json
│   ├── events.json
│   ├── grafana-847888c644-vj8qz
│   │   └── logs.txt
│   ├── heapster-5db74f4bbd-5t6qx
│   │   └── logs.txt
│   ├── influxdb-78f4754b8f-zql59
│   │   └── logs.txt
│   ├── kapacitor-84c57d6cb-qpkt2
│   │   └── logs.txt
│   ├── nethealth-9dg7b
│   │   └── logs.txt
│   ├── nethealth-fz7pd
│   │   └── logs.txt
│   ├── nethealth-xgrkm
│   │   └── logs.txt
│   ├── pods.json
│   ├── replicasets.json
│   ├── replication-controllers.json
│   ├── services.json
│   ├── telegraf-7bfc97698c-s6rc6
│   │   └── logs.txt
│   ├── telegraf-node-master-rsm44
│   │   └── logs.txt
│   ├── telegraf-node-master-s7gq8
│   │   └── logs.txt
│   └── telegraf-node-master-xtzfn
│       └── logs.txt
└── nodes.json
[...]
```

* --since=1m
```
├── k8s-logs-kube-system-gravity-site-c7jxz-gravity-site
├── k8s-logs-kube-system-gravity-site-xlrwl-gravity-site
├── k8s-logs-kube-system-gravity-site-zg7j5-gravity-site
├── k8s-logs-kube-system-site-app-post-install-2bcd3f-79kft-post-install-hook-prev
├── k8s-logs-monitoring-influxdb-78f4754b8f-zql59-influxdb
├── k8s-logs-monitoring-kapacitor-84c57d6cb-qpkt2-kapacitor
[...]
```

Reply:
Besides the logs, I think it also collects the dump of all resources? IIRC we already collect those too (kubectl describe outputs), so I don't think we need to do anything separately.